### ISSUE TITLE
Java method calls are ignored in generating call/caller graph with Graphviz

### ISSUE THREAD

**Issue by Nakata328** [doxygen/doxygen#6469]
If a variable is declared just after instantiation of class with generics, such as List or Map,
method calls of the variable are ignored in generating call/caller graph with Graphviz.
(i.e. relations between two methods are not figured in picture)

In example below,
•MyCaller#executeOK->MyCalled#execute(MyCaller.java:11) is generated.
•MyCaller#executeNG->MyCalled#execute(MyCaller.java:17) isn't generated.

[MyCaller.java]--------------------------------------
```
01 package mysample.caller;
02
03 import java.util.ArrayList;
04 import java.util.List;
05 import mysample.called.MyCalled;
06
07 public class MyCaller {
08
09     public void executeOK() {
10         MyCalled called = new MyCalled();
11         called.execute();
12     }
13
14     public void executeNG() {
15         List<String> list = new ArrayList<>();
16         MyCalled called = new MyCalled();
17         called.execute();
18         list.add("");
19     }
20 }

```
[MyCalled.java]--------------------------------------
```
01 package mysample.called;
02 
03 public class MyCalled {
04     public void execute() {}
05 }
```
-----------------------------------------------------

![screenshot](https://user-images.githubusercontent.com/42762666/44705304-c2cb1580-aa8d-11e8-8946-019037f52aef.jpg)

If I insert a some code into between line 15 and 16,
MyCaller#executeNG->MyCalled#execute(MyCaller.java:17) is also generated.

My enviroment:
•doxygen-1.8.14
•graphviz-2.38
•Windows7 32bit

•Configured parameters:
  - OUTPUT_DIRECTORY
  - INPUT
  - DOT_PATH
  - DOTFILE_DIRS

•Change parameters to YES from NO:
  - RECURSIVE
  - HAVE_DOT
  - CALL_GRAPH
  - CALLER_GRAPH
  - EXTRACT_ALL

**Comment by albert-github** [comment]
Looks like the `<>` is the problem, Is this a newer feature of Java?

Interchanging lines 15 and 16 in the above makes it working (this is just a workaround).

**Comment by Nakata328** [comment]
"new ArrayList<>()" is notation valid from Java7.
But even if changing it to "new ArrayList<String>()", notation valid in Java6 or earlier, the same issue happens.

Besides the way you said, changing it to "new ArrayList()" also makes it working, but it is deprecated Java code.
If possible, I want to avoid this issue without changing Java source.

**Comment by albert-github** [comment]
I thought my tests showed that `new ArrayList()` did work, but this is by head ( a bit of a contradiction in your comment as first you say "the same happens" and later say "also makes it working".

The deprecated remark is this only foe the `new ArrayList()` change or also for the order of the statements (the later would wonder me.

**Comment by Nakata328** [comment]
Sorry, there was a typo.

[Wrong] But even if changing it to "new ArrayList()", notation valid in Java6 or earlier, ....
[Right] But even if changing it to "new ArrayList<String>()", notation valid in Java6 or earlier, ....

**Comment by Nakata328** [comment]
Sorry, there was a mistake.

[Wrong] But even if changing it to "new ArrayList()", notation valid in Java6 or earlier, ....
[Right] But even if changing it to "new ArrayList< String >()", notation valid in Java6 or earlier, ....

("<" + "String" + ">" is deleted when I comment, so insert space before and behind "String")

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.15.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
